applications/GeoMechanicsApplication/custom_utilities/math_utilities.cpp

```diff
@@ -15,11 +15,31 @@
 #include "math_utilities.h"
 #include "utilities/math_utils.h"
 
-std::vector<double> Kratos::GeoMechanicsMathUtilities::CalculateDeterminants(const std::vector<Matrix>& rMatrices)
+namespace Kratos
+{
+
+std::vector<double> GeoMechanicsMathUtilities::CalculateDeterminants(const std::vector<Matrix>& rMatrices)
 {
     std::vector<double> result(rMatrices.size());
     std::transform(rMatrices.cbegin(), rMatrices.cend(), result.begin(),
                    [](const auto& rMatrix) { return MathUtils<>::Det(rMatrix); });
 
     return result;
 }
+
+Matrix GeoMechanicsMathUtilities::VectorToDiagonalMatrix(const Vector& rVector)
+{
+    Matrix result = ZeroMatrix(rVector.size(), rVector.size());
+    for (std::size_t i = 0; i < rVector.size(); ++i) {
+        result(i, i) = rVector(i);
+    }
+    return result;
+}
+
+Matrix GeoMechanicsMathUtilities::RotateTensor(const Matrix& rTensor, const Matrix& rRotationMatrix)
+{
+    Matrix temp = prod(rTensor, trans(rRotationMatrix));
+    return prod(rRotationMatrix, temp);
```

Comment:
We can rewrite this as a one-liner:
```suggestion
    return prod(rRotationMatrix, Matrix{prod(rTensor, trans(rRotationMatrix))});
```